atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/ManyClientPoolingContainer.java

```diff
@@ -117,8 +121,8 @@ public synchronized void shutdownPooling() {
 
     @Override
     protected PoolingContainer<Client> delegate() {
-        List<PoolingContainer<Client>> list = containers;
-        return list.get(random.nextInt(list.size()));
+        ImmutableList<PoolingContainer<Client>> list = containers; // volatile read
```

Comment:
aaaaand I missed the 2nd reference to list. I'll make sure to make all future PR comments after 10am.